auction-server/src/state.rs

```diff
@@ -472,16 +495,121 @@ pub struct ChainStoreEvm {
     pub name:                   String,
 }
 
+impl ChainStoreEvm {
+    pub fn get_chain_provider(
+        chain_id: &String,
+        chain_config: &ConfigEvm,
+    ) -> anyhow::Result<Provider<TracedClient>> {
+        let mut provider = TracedClient::new(
+            chain_id.clone(),
+            &chain_config.geth_rpc_addr,
+            chain_config.rpc_timeout,
+        )
+        .map_err(|err| {
+            tracing::error!(
+                "Failed to create provider for chain({chain_id}) at {rpc_addr}: {:?}",
+                err,
+                chain_id = chain_id,
+                rpc_addr = chain_config.geth_rpc_addr
+            );
+            anyhow!(
+                "Failed to connect to chain({chain_id}) at {rpc_addr}: {:?}",
+                err,
+                chain_id = chain_id,
+                rpc_addr = chain_config.geth_rpc_addr
+            )
+        })?;
+        provider.set_interval(Duration::from_secs(chain_config.poll_interval));
+        Ok(provider)
+    }
+    pub async fn create_store(
```

Comment:
I think this function is like the `new` function inside the svm chain_store. Can we add these to `ChainStore` trait?
Maybe we can have sth like `try_new` or `create_store` for the ChainStore trait. wdyt?

Reply:
I don't think it's worth the abstraction. We know if we are creating Svm version or an Evm version right now but if at some point we have a function that's creating both of them without a lot of distinguishing, we can unify them for sure.